### PR TITLE
List of current changes: 

### PR DESCRIPTION
**Pull request by IgorKryviuk**
1.	Organise single class imports.
   2.	Improve CartDAO and HistoryDAO classes.
   3.	Add CartController and CartService.
   4.	Add HistoryController and HistoryService.

### CODE REVIEW

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public void insert(History history) {
+        historyDAO.insert(history);
+        LOGGER.info(String.format(successfullyOperation, history.getProduct().getName(), "inserted into"));
+    }
+
+    public void update(History history){
+        if (historyDAO.update(history) == 0) {
+            throw new IllegalArgumentException(String.format(historyNotFound, "invalid UPDATE operation", history.getProduct().getName()));
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java]: To be consistent with the whole application better to throw our custom DataNotFoundException

File: RESTserver/src/main/java/com/softserve/if072/restservice/controller/HistoryController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    @ExceptionHandler(IllegalArgumentException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public String dataNotFound(IllegalArgumentException e) {
+        LOGGER.error(e.getMessage());
+        return e.getMessage();
+    }
+
+    @ExceptionHandler(DataAccessException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public String dataAccessException(DataAccessException e) {
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/controller/HistoryController.java]: To not duplicate this in all controllers, please create super class, move these two methods there and extends it by other contollers.

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java (hunk 1 above)
Comment (IgorKryviuk) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java]: I've done

File: RESTserver/src/main/java/com/softserve/if072/restservice/controller/HistoryController.java (hunk 2 above)
Comment (IgorKryviuk) [inline RESTserver/src/main/java/com/softserve/if072/restservice/controller/HistoryController.java]: I've done

**Pull request merged**